bots/tests-scan

```diff
@@ -270,17 +296,52 @@ def cockpit_tasks (api, update, policy):
             else:
                 (priority, changes) = prioritize(status, title, labels, baseline, context)
             if not update or update_status(api, revision, context, status, changes):
-                results.append((priority, "pull-%d" % number, revision, "pull/%d/head" % number, context, base))
+                results.append((priority, "pull-%d" % number, revision, "pull/%d/head" % number, context, base, repo, None))
+
+        # if a cockpit (default repo) PR changes bots/, also trigger all external tests against this cockpit PR
+
+        def trigger_externals():
+            if repo:  # already a non-cockpit project
+                return False
+            if TRIGGER_EXTERNAL in labels():  # already checked before?
+                return True
+
+            if not have:
+                # this is the first time tests-scan looks at a PR, so determine if it changes bots/
+                with urllib.request.urlopen(pull["patch_url"]) as f:
+                    # enough to look at the git commit header, it lists all changed files
+                    if b"bots/" in f.read(4000):
```

Comment:
This should also be ```src/``` ... Any changes to ```src/``` can be expected to affect external projects.

Reply:
They don't right now. The only thing external projects check out from cockpit is bots/ (and tests/, but they pin that to a particular tag, so testing PRs won't change things). External projects install the distribution cockpit packages. Perhaps in the future we can build VMs with cockpit from a cockpit PR and somehow inject that into the test suite of starter-kit-like projects, but there are still some steps to go for that. So right now it doesn't make sense to include src/.

Reply:
Makes sense. I guess that's a follow up? To install a custom Cockpit when we change src/ ?

Reply:
Maybe some day, but not immediately after. This requires a lot of infrastructure work, thinking about how to store and pass VMs between bots or how to test multiple projects in one test run, and is generally rather hairy. Also, the benefits of that are relatively marginal - we have to ensure we don't break Cockpit's API, and we must do that already within Cockpit (like we currently do with the rhel-7-5 image). If we break the API and don't notice, and rely on external project tests to tell us, we have already failed IMHO.

Reply:
Okay, but we currently use these exact same tests to tell us when we break API ... and we are moving those tests external. So we are increasing the gap. We currently rely on ourselves to figure out that we broke API, and we're diluting that capability.

Lets be very aware of this, and perhaps increase validation of our API?